burn-tensor/src/tensor/ops/activation.rs

```diff
@@ -8,6 +9,48 @@ use super::FloatTensor;
 ///
 /// This trait let backend implementations override activation functions for better performance.
 pub trait ActivationOps<B: Backend> {
+    /// Applies the LeakyReLU activation function.
+    ///
+    /// # Arguments
+    ///
+    /// * `tensor` - The tensor.
+    /// * `alpha` - The alpha value that values smaller than 0 are multiplied with.
+    ///
+    /// # Returns
+    ///
+    /// The output tensor.
+    fn leaky_relu<const D: usize>(
+        tensor: FloatTensor<B, D>,
+        alpha: super::FloatElem<B>,
+    ) -> FloatTensor<B, D> {
+        let mask = B::float_lower_elem(tensor.clone(), 0.elem());
+        let ones = B::float_ones(Shape { dims: [1; D] }, &B::float_device(&tensor));
+        let alpha_tensor = B::float_full(Shape { dims: [1; D] }, alpha, &B::float_device(&tensor));
+        let mul_mask = B::float_mask_where(ones, mask, alpha_tensor);
+        B::float_mul(tensor, mul_mask)
```

Comment:
This seems wrong, the shapes of `ones` and `alpha` should be the same as the inputs.

```rust
        let mask = B::float_lower_elem(tensor.clone(), 0.elem());
        let alpha_tensor = B::float_mul(tensor.clone(), alpha.elem());
        
        // Update the tensor where the values are `> 0` by `tensor * alpha`.
        B::float_mask_where(tensor, mask, alpha_tensor)
```

Reply:
Done, but used float_mul_scalar